# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.cpp
# Commit: c93972b05d4f
# Full Hash: c93972b05d4f105953213185bd4e0da2403dedd1
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Mark mCapturedFrameInvalidated even if mIsCanvasDirty. r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79014
# ==============================================================================

diff -r 04f5127c85d5 -r c93972b05d4f dom/canvas/WebGLContext.cpp
--- a/dom/canvas/WebGLContext.cpp	Wed Jun 10 22:21:17 2020 +0000
+++ b/dom/canvas/WebGLContext.cpp	Wed Jun 10 22:21:19 2020 +0000
@@ -248,13 +248,13 @@
 }
 
 void ClientWebGLContext::MarkCanvasDirty() {
-  if (mIsCanvasDirty) return;
-  mIsCanvasDirty = true;
-
   if (!mCanvasElement) return;
 
   mCapturedFrameInvalidated = true;
 
+  if (mIsCanvasDirty) return;
+  mIsCanvasDirty = true;
+
   SVGObserverUtils::InvalidateDirectRenderingObservers(mCanvasElement);
 
   mCanvasElement->InvalidateCanvasContent(nullptr);
